src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeEventDispatcher.cs

```diff
@@ -179,6 +187,11 @@ private unsafe void DispatchEventsToEventListeners()
                 // Wait for more events.
                 if (!m_stopDispatchTask)
                 {
+                    if (!eventsReceived)
+                    {
+                        waitHandle.WaitOne();
```

Comment:
When the EventListener stops listening, what wakes this up?

Reply:
Right now nothing. I verified that if you wait for the listener to go to sleep and then try to disable events it hangs forever. I'll have a fix shortly